yb-voyager/cmd/common.go

```diff
@@ -479,17 +480,68 @@ func initMetaDB(migrationExportDir string) *metadb.MetaDB {
 	if err != nil {
 		utils.ErrExit("get migration status record: %v", err)
 	}
-	if msr.VoyagerVersion != utils.YB_VOYAGER_VERSION {
-		userFacingMsg := fmt.Sprintf("Voyager requires the entire migration workflow to be executed using a single Voyager version.\n"+
-			"The export-dir %q was created using version %q and the current version is %q. Either use Voyager %q to continue the migration or start afresh "+
-			"with a new export-dir.", migrationExportDir, msr.VoyagerVersion, utils.YB_VOYAGER_VERSION, msr.VoyagerVersion)
+
+	// If the msr VoyagerVersion is less than the PREVIOUS_BREAKING_CHANGE_VERSION, then the export-dir is not compatible with the current Voyager version.
+	// This version will always be a final release version and never "main" or "rc" version.
+	previousBreakingChangeVersion, err := version.NewVersion(utils.PREVIOUS_BREAKING_CHANGE_VERSION)
+	if err != nil {
+		utils.ErrExit("could not create version from %q: %v", utils.PREVIOUS_BREAKING_CHANGE_VERSION, err)
+	}
+
+	var versionCheckFailed bool
+
+	if msr.VoyagerVersion == "main" {
+		// If the export-dir was created using the main branch, then the current version should also be the main branch.
+		if utils.YB_VOYAGER_VERSION != "main" {
+			versionCheckFailed = true
+		}
+	} else if msr.VoyagerVersion != "" {
+		msrVoyagerVersionString := msr.VoyagerVersion
+		if strings.Contains(msrVoyagerVersionString, "rc") {
+			// RC version will be like 0rc1.1.8.6
```

Comment:
maybe have a utility function for extracting final release from RC format

Reply:
Done